gax/src/main/java/com/google/api/gax/rpc/ApiCallContext.java

```diff
@@ -153,6 +156,54 @@
   @BetaApi("The surface for tracing is not stable yet and may change in the future")
   ApiCallContext withTracer(@Nonnull ApiTracer tracer);
 
+  /**
+   * Returns a new ApiCallContext with the given {@link RetrySettings} set.
+   *
+   * <p>This sets the {@link RetrySettings} to use for the RPC. These settings will work in
+   * combination with either the default retryable codes for the RPC, or the retryable codes
+   * supplied through {@link #withRetryableCodes(Set)}. Calling {@link
+   * #withRetrySettings(RetrySettings)} on an RPC that does not include {@link
+   * Code#DEADLINE_EXCEEDED} as one of its retryable codes (or without calling {@link
+   * #withRetryableCodes(Set)} with a set that includes at least {@link Code#DEADLINE_EXCEEDED})
+   * will effectively only set a simple timeout that is equal to {@link
+   * RetrySettings#getInitialRpcTimeout()}. It is recommended to use {@link #withTimeout(Duration)}
+   * if that is the intended behavior.
```

Comment:
I'm actually not sure that `withTimeout(Duration)` is the recommended API to use. We've discussed this in https://github.com/googleapis/gax-java/issues/1144#issuecomment-663663167, but the [code is pretty explicit](https://github.com/googleapis/gax-java/blob/27d92c6cf54d3fb4ba94bf6cd95e57ea9e921e45/gax/src/main/java/com/google/api/gax/rpc/AttemptCallable.java#L72-L75) in how it overwrites the `ApiCallContext.withTimeout` via the `attemptSettings` (the RPC timeout calculated by a `RetryAlgorithm`, like `ExponentialRetryAlgorithm`). Unfortunately, this public API `withTimeout` isn't really usable for _retryable_ RPCs.

Reply:
FWIW I think we could fix that very easily and we could allow users to set a "constant, per-RPC" timeout via `withTimeout`.

Reply:
Opened https://github.com/googleapis/gax-java/pull/1324 in case we want to fix it.

Reply:
We ended up fixing `withTimeout` with that PR BTW.